c++/src/kj/async-inl.h

```diff
@@ -2247,6 +2272,15 @@ class Coroutine final: public CoroutineBase,
   // a type-erased `await_suspend(stdcoro::coroutine_handle<void>)` override, and implement
   // suspension and resumption in terms of .then(). Yuck!
 
+  template <typename U>
+  class ForkedPromiseAwaiter;
+
+  // called by co_awaiting on a forked promise.
+  template <typename U>
+  ForkedPromiseAwaiter<U> await_transform(ForkedPromise<U>&& promise) { 
```

Comment:
I think the idea is that this should take an lvalue reference, right?

Otherwise the ForkedPromise can only be `co_await`ed once, which defeats the purpose of `ForkedPromise`.

Reply:
the idea was actually to have rvalue, and await on the promise once all the branches are already established:

```
auto fork = promise.fork();
auto branch = promise.branch();
...

co_await kj::mv(fork);
```

Reply:
but I can easily redo this as an rvalue if you think it would be better.

Reply:
I believe harris's idea was in fact to accept an lvalue reference.

It doesn't seem like there's any benefit to requiring an rvalue here? We could accept an lvalue with exactly the same implementation, but it would then allow co_awating the forked promise multiple times.

Reply:
done.